cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -109,13 +115,97 @@ func (spm *KubeStaticPodPathManager) BackupManifestDir() string {
 	return spm.backupManifestDir
 }
 
-// StaticPodControlPlane upgrades a static pod-hosted control plane
-func StaticPodControlPlane(waiter apiclient.Waiter, pathMgr StaticPodPathManager, cfg *kubeadmapi.MasterConfiguration) error {
+func performStaticPodUpgrade(component string, waiter apiclient.Waiter, pathMgr StaticPodPathManager, cfg *kubeadmapi.MasterConfiguration, beforePodHash string, recoverManifests map[string]string) error {
```

Comment:
no logic changes here, right?
Just func extraction?

Reply:
Logic changes with regards to processing, previous func was upgrading all components, new func updates one component at the time.